### ISSUE TITLE
Unbuildable libiothsm-std in releases since (not including) 1.1.11

### ISSUE THREAD

**Issue by sindelar-fr** [Azure/iotedge#6525]
I'd log this under azure-iotedge, but it seems impossible ... 

I'm trying to move the meta-iotedge to 1.1.15 and I ran in to an issue trying to move everything to 1.1.15. I'm unable to build libiothsm-std 1.1.15. 

It seems that the release tarball is missing the content of iotedge-<version>/hsm-sys/azure-iot-hsm-c/deps/c-shared and iotedge-<version>/hsm-sys/azure-iot-hsm-c/deps/utpm is missing in the release tarballs of iotedge since version 1.1.11, 1.1.11 being the last OK release that I am able to build.

I'm not sure why since these don't seem to have changed for about 15 months.

**Comment by yophilav** [comment]
@sindelar-fr 
Thank you for the report! We found the issue. Working on the fix right now.

**Comment by sindelar-fr** [comment]
Thanks!
Maybe it would be a good idea to base the meta-iotedge on git rather than tarball ...

**Comment by yophilav** [comment]
Unfortunately it's not quite that simple. The tarball contains all the source code for deps required to build iotedge and libiothsm -- the source code of the deps are not contained in the /meta-iotedge.

**Comment by sindelar-fr** [comment]
Sure - they are here in this repo as submodules, perhaps some licensing issue? Anyway this is not quite the right place... :)

**Comment by yophilav** [comment]
@sindelar-fr 
Thank you for your patience! Our build pipeline is broken for /meta-iotedge and the contributor who has access to it is OOF. I have a draft PR out that I have yet to test here:  https://github.com/Azure/meta-iotedge/pull/85/files. Feel free to give it a try and provide me feedback if you see further issues :)

In the meantime, let me ask around if I can have the permission to the github repo so I can fix the pipeline.

**Comment by sindelar-fr** [comment]
Hmm maybe it's misleading that I've writtend what I was trying to do.
The issues is not moving meta-iotedge to 1.1.15 - for that project I am able to create issues.
The issue is this:
![image](https://user-images.githubusercontent.com/131410/180595753-0ddb887a-6fca-420b-8888-8040c4309891.png)
I don't think that build using your changes in meta-iotedge will work. I'd say it will fail just as it did for me, because the [release tarball ](https://github.com/Azure/azure-iotedge/releases/download/1.1.15/iotedge-1.1.15.tar.gz) is missing the content of those two directories. (All releases > 1.1.11 are missing the content of those directories)

It doesn't seem possible to log issues against https://github.com/Azure/azure-iotedge, so I've logged it here, but the problem is in https://github.com/Azure/azure-iotedge.

I currently don't have access to the right computer to verify my statement about the build failure of your PR in meta-iotedge, maybe sometime next week.

**Comment by emilm** [comment]
@yophilav 

My libiothsm-std_1.1.15.bb was exactly the same as yours and I am experiencing the same errors  as @sindelar-fr:
```
| CMake Error at CMakeLists.txt:132 (add_subdirectory):
|   The source directory
|
|     .../build/tmp-glibc/work/cortexa7t2hf-neon-vfpv4-oe-linux-gnueabi/libiothsm-std/1.1.15-r0/iotedge-1.1.15/hsm-sys/azure-iot-hsm-c/deps/c-shared
|
|   does not contain a CMakeLists.txt file.
|
|
| CMake Error at CMakeLists.txt:133 (add_subdirectory):
|   The source directory
|
|     .../build/tmp-glibc/work/cortexa7t2hf-neon-vfpv4-oe-linux-gnueabi/libiothsm-std/1.1.15-r0/iotedge-1.1.15/hsm-sys/azure-iot-hsm-c/deps/utpm
|
|   does not contain a CMakeLists.txt file.
|
|
| -- Configuring incomplete, errors occurred!
| See also ".../build/tmp-glibc/work/cortexa7t2hf-neon-vfpv4-oe-linux-gnueabi/libiothsm-std/1.1.15-r0/build/CMakeFiles/CMakeOutput.log".
| WARNING: exit code 1 from a shell command.

```

**Comment by yophilav** [comment]
@sindelar-fr / @emilm 
Thank you for the clarification! Taking a look...

**Comment by yophilav** [comment]
@sindelar-fr / @emilm 
Appreciate your patient. Could you try 
https://github.com/Azure/azure-iotedge/releases/download/1.1.15/iotedge-1.1.15.tar.gz
against
https://github.com/Azure/meta-iotedge/pull/85
please?

**Comment by sindelar-fr** [comment]
I've tried my patched recipes for 1.1.15 and they all passed. If you are going forward with https://github.com/Azure/meta-iotedge/pull/85, please patch also the recipe for iotedge-cli so that it's all 1.1.15.

**Comment by emilm** [comment]
Works here too!
